cpp/cgi_progs/translator_statistics.cc

```diff
@@ -0,0 +1,209 @@
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <algorithm>
+#include <iostream>
+#include <set>
+#include <sstream>
+#include <string>
+#include "DbConnection.h"
+#include "DbResultSet.h"
+#include "DbRow.h"
+#include "EmailSender.h"
+#include "HtmlUtil.h"
+#include "IniFile.h"
+#include "StringUtil.h"
+#include "Template.h"
+#include "UBTools.h"
+#include "UrlUtil.h"
+#include "util.h"
+#include "WebUtil.h"
+
+
+namespace {
+
+const std::string CONF_FILE_PATH(UBTools::GetTuelibPath() + "translations.conf");
+
+enum Category { VUFIND, KEYWORDS };
+
+
+DbResultSet ExecSqlAndReturnResultsOrDie(const std::string &select_statement, DbConnection * const db_connection) {
+    db_connection->queryOrDie(select_statement);
+    return db_connection->getLastResultSet();
+}
+
+
+// std::vector<std::string> GetLanguageCodesFromTable(DbConnection &db_connection, const std::string &table_name) {
+//     const std::string query("SELECT DISTINCT language_code from " + table_name + " ORDER BY language_code;");
+//     DbResultSet result_set(ExecSqlAndReturnResultsOrDie(query, &db_connection));
+//     std::vector <std::string> language_codes;
+//     while (const DbRow db_row = result_set.getNextRow())
+//         language_codes.emplace_back(db_row["language_code"]);
+
+//     return language_codes;
+// }
+
+
+// std::vector<std::string> GetLanguageCodes(DbConnection &db_connection) {
+//     std::vector <std::string> language_codes(GetLanguageCodesFromTable(db_connection, "vufind_translations"));
+//     for (auto &language_code : GetLanguageCodesFromTable(db_connection, "keyword_translations")) {
+//         if (std::find(language_codes.begin(), language_codes.end(), language_code) == language_codes.end())
+//             language_codes.emplace_back(language_code);
+//     }
+//     return language_codes;
+// }
+
+
+// void ShowErrorPageAndDie(const std::string &title, const std::string &error_message, const std::string &description = "") {
+//     std::cout << "Content-Type: text/html; charset=utf-8\r\n\r\n";
+//     std::cout << "<!DOCTYPE html><html><head><title>" + title + "</title></head>"
+//               << "<body>"
+//               << "  <h1>" + error_message + "</h1>"
+//               << "  <h3>" + description + "</h3>"
+//               << "</body>"
+//               << "</html>";
+//     std::exit(EXIT_SUCCESS);
+// }
+
+
+
+std::string GetCGIParameterOrDefault(const std::multimap<std::string, std::string> &cgi_args,
```

Comment:
I think that this kind of function is copy/pasted through multiple CGI programs... can we move this one to WebUtil?